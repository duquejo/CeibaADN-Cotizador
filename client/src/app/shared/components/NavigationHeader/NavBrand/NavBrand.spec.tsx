import * as React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import LogoCeiba from 'assets/img/logo-ceiba.png';
import { NavBrand } from './index';

describe('Pruebas para el componente <NavBrand />', () => {

    let componentWrapper: ShallowWrapper;

    afterEach(() => {
        componentWrapper.unmount();
    });


    it('Debería renderizar el logo de la aplicación', () => {
        // Arrange
        const params = {
            text: 'Demo',
            imgSrc: LogoCeiba
        };       
        // Act
        componentWrapper = shallow( 
            <NavBrand
                { ...params }
            />
        );
        // Assert
        expect( componentWrapper ).toMatchSnapshot();
        expect( componentWrapper.find('LogoImg').prop('src') ).toBe(params.imgSrc);
    });

    it('No debería renderizar el logo de la aplicación, sino el texto si no está presente el parámetro', () => {
        // Arrange
        const params = {
            text: 'Demo'
        };       
        // Act
        componentWrapper = shallow( 
            <NavBrand
                { ...params }
            />
        );
        // Assert
        expect( componentWrapper ).toMatchSnapshot();
        expect( componentWrapper.find('LogoSpan').text() ).toBe(params.text);
    });
});
