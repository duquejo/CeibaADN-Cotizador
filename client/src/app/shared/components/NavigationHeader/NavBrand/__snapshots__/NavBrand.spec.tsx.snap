// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <NavBrand /> Debería renderizar el logo de la aplicación 1`] = `
<BrandDiv>
  <Styled(Link)
    replace={true}
    to="/"
  >
    <LogoImg
      alt="Demo"
      src="logo-ceiba.png"
    />
  </Styled(Link)>
</BrandDiv>
`;

exports[`Pruebas para el componente <NavBrand /> No debería renderizar el logo de la aplicación, sino el texto si no está presente el parámetro 1`] = `
<BrandDiv>
  <Styled(Link)
    replace={true}
    to="/"
  >
    <LogoSpan>
      Demo
    </LogoSpan>
  </Styled(Link)>
</BrandDiv>
`;
